omniport/core/base_auth/managers/get_user.py

```diff
@@ -56,20 +56,20 @@ def get_user(username):
         q_institute_webmail_address = Q(institute_webmail_address=username)
         q = (
                 q_primary_phone_number
-                | q_secondary_phone_number
                 | q_email_address
                 | q_institute_webmail_address
         )
-        contact_information = ContactInformation.objects.get(q)
+        entity_supported = ContentType.objects.get_for_model(Person)
+        try:
+            contact_information = ContactInformation.objects.filter(entity_content_type=entity_supported).get(q)
+        except ContactInformation.MultipleObjectsReturned:
+            contact_information = ContactInformation.objects.filter(entity_content_type=entity_supported).get(institute_webmail_address=username)
         entity = contact_information.entity
-        if type(entity) is Person:
-            user = entity.user
-            if user is not None:
```

Comment:
This is an important condition to check.

Reply:
Line number 62 handles this automatically its just redundant to check here once again